lib/cfndsl/Types.rb

```diff
@@ -45,12 +39,15 @@ class AWSType < JSONable
     
     # Go through and declare all of the types first
     aws_types["Types"].each_key do |typename|
-      if( ! AWSTypes.const_defined? typename ) then
-        klass = AWSTypes.const_set( typename, Class.new(AWSType ) )
-        classes[typename] = klass
-      else
-        classes[typename] = AWSTypes.const_get(typename)
-      end 
+      next if %w(String JSON Integer).include?(typename)
+
+      if AWSTypes.const_defined? typename then
+        puts "duplicate type #{typename}"
```

Comment:
All the error checking code above was basically doing the same thing so I replicated it.
I think we should probably replace all the ```puts``` in this file with a ```fail``` perhaps?

Since it would mean aws_types.yaml is broken and we would then pick it up in the tests and fix it before releasing a new version?